### ISSUE TITLE
win_iis_website ssl flag does nothing

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#25465]
_From @nathanwebsterdotme on August 17, 2016 12:40_

##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

win_iis_website
##### ANSIBLE VERSION

```
ansible 2.1.1.0 (detached HEAD 35da6ba9d1) last updated 2016/07/25 09:38:23 (GMT +000)
  lib/ansible/modules/core: (detached HEAD 45128c8bab) last updated 2016/07/25 09:38:23 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD 511752e53a) last updated 2016/07/25 09:38:23 (GMT +000)
  config file = ********
  configured module search path = ['modules']
```
##### OS / ENVIRONMENT

Server 2012 R2
##### SUMMARY

The ssl flag on web_iis_website doesn't do anything.
##### STEPS TO REPRODUCE

Run the win_iis_webssite module with 'ssl' option set to true. 

```
    - name: IIS | Create Website with SSL Binding
      win_iis_website:
        name: "SSLSite"
        physical_path: "C:\Websites\SSLSite"
        application_pool: "SSLSite-AppPool"
        hostname: "SSLsite.test.com"
        state: "present"
        ssl: true
```

##### EXPECTED RESULTS

SSL binding attached to website upon creation
##### ACTUAL RESULTS

No SSL binding is created on IIS.

_Copied from original issue: ansible/ansible-modules-extras#2746_

**Comment by dagwieers** [comment]
_From @ansibot on August 17, 2016 12:50_



@whenrik ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @deekayen on August 23, 2016 22:3_

I've been trying to get SSL to bind using 2.1.1.0 and this report looks like it may be my scapegoat for failure. At first, it was merely peculiar that the docs didn't have any choices for the ssl argument, so picking true as a value for ssl was just a guess. When I tracked down the `win_iis_website.ps1` file, it doesn't appear to do anything with the `ssl` param.

```
$ pwd
/usr/lib/python2.7/site-packages/ansible
$ grep -r bind_ssl *
modules/extras/windows/win_iis_website.ps1:$bind_ssl = Get-Attr $params "ssl" $FALSE;
```

It just gets ssl attributes, assigns to a `$bind_ssl` variable, and then that variable never appears to be used in any code in all of the ansible install directory.

<img width="660" alt="Site bindings with http on port 443" src="https://cloud.githubusercontent.com/assets/382319/17910797/75f2f926-6958-11e6-9c5c-f98510a48182.png">

The type value on that should say `https`, not `http`. It makes sort of a chicken-egg problem for me because the playbook fails when I try to bind port 443 using the http type, so I can't use the webbind module to do the ssl fix instead since the website has to exist to edit the binding.

<img width="498" alt="Port is already in use error." src="https://cloud.githubusercontent.com/assets/382319/17910875/ca037a22-6958-11e6-8401-28d7443b5b21.png">

I have a different website running on port 80 to redirect to 443.

If I manually go back to the website bindings in IIS Manager, change the port of the http type to 80 on the failing website task, then add https with port 443, then remove the http type, IIS Manager doesn't complain with the port in-use error anymore. Then I can rereplay the playbook to finish out my tasks.

**Comment by dagwieers** [comment]
_From @nathanwebsterdotme on September 5, 2016 13:29_

I'm currently having to add an SSL binding as a separate task using win_iis_webbinding

**Comment by dagwieers** [comment]
_From @ansibot on September 5, 2016 13:29_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @nathanwebsterdotme on September 15, 2016 13:57_

Any updates?

**Comment by dagwieers** [comment]
_From @ansibot on October 6, 2016 17:50_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on October 22, 2016 10:8_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on November 6, 2016 10:44_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on November 22, 2016 10:51_

@whenrik, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
_From @ansibot on December 6, 2016 23:7_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
cc @jborean93 @jhawkesworth
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
just an FYI, I'm planning on setting the binding parameters in `win_iis_website` to be deprecated in favour of using `win_iis_webbindings`. This is to keep the modules simpler to deal with as bindings can get quite complex.

**Comment by daniyalj** [comment]
I am running into the same issue. Having the SSL flag on `win_iis_webbindings` makes much more sense. Any update on this? @jborean93

**Comment by ansibot** [comment]
cc @nitzmahone
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @if-meaton
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ShachafGoldstein** [comment]
@jborean93 - Can this issue be closed?

**Comment by unix-bomber** [comment]
Receiving this error. Using win_iis_webbinding as a workaround. Perhaps remove ssl as win_iis_webbinding is effective enough?

**Comment by jborean93** [comment]
Due to the age of this issue I am closing this issue. If there is still a specific problem please raise it at https://github.com/ansible-collections/community.windows which is where the module is now located.
